01_easy_algorithms/04_sharing_is_caring/04_sharing_is_caring.rb

```diff
@@ -0,0 +1,4 @@
+def show_the_love(x)
+  diff = x.map { |n| n*0.25 unless n == x.min }.compact!.sum
+  x.map { |n| n*0.75 unless n == x.min }.map { |n| n == nil ? x.min+diff : n}
+end
```

Comment:
The way the code is written is too confusing and requires more time for another developer to figure it out (or you when you revisit this code after a while).

Remember, the best code isn't necessarily the shortest. It's the code that is most easily understood.

So restructure this so that it's more readable and follows a logical order.